src/OrbitGl/FormatCallstackForTooltipTest.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_join.h>
+#include <gtest/gtest.h>
+
+#include "ClientData/ModuleAndFunctionLookup.h"
+#include "FormatCallstackForTooltip.h"
+
+using orbit_client_data::CallstackInfo;
+using orbit_client_data::CallstackType;
+using orbit_client_data::CaptureData;
+using orbit_client_data::LinuxAddressInfo;
+using orbit_client_data::ModuleManager;
+using orbit_grpc_protos::CaptureStarted;
+
+namespace orbit_gl {
+
+namespace {
+constexpr uint64_t kFrame1 = 0xADD5E55;
+constexpr uint64_t kOffsetInFunction = 0x12;
+constexpr const char* kModulePath = "/path/to/module";
+constexpr const char* kModuleName = "module";
+constexpr const char* kFunctionNameWithSpecialChars = "void foo<int>(const Foo&)";
+constexpr const char* kEscapedFunctionName = "void foo&lt;int&gt;(const Foo&amp;)";
+}  // namespace
+
+TEST(CallstackTooltipUtils,
+     FormatInnermostFrameOfEmptyCallstackForTooltipYieldsUnkownModuleAndFunctionName) {
```

Comment:
Long test names are not a problem. It's annoying not to have spaces or punctuation but other than that I don't see the problem.

Regarding the rest, I don't have a strong opinion and there are different opinions around. But my non-strong opinion is that I'm also not a big fan of the huge code duplication in tests: while a unit test should test a single unit because it's easier to immediately see what broke, the benefit of immediately seeing what broke rather than having to look for a few seconds is basically none (it's not like the test breaks all the time). Also `EXPECT_EQ` tells you the exact line anyway. But on the other hand, if you ever have to change the tests, having less code to change and keep in sync across tests saves you a lot of time.

So I think in this case a set of tests that test both functions at the same time on the same setup is reasonable. For example this test can simply be called "EmptyCallstack" and the other "HtmlEscaping". Otherwise, the common code could be extracted into a function, but that's definitely already less convenient.

Reply:
I am not convinced. We are talking about saving 3 lines of code in a test. Lines that are very simple to read. We already extract constants shared across tests, so that it makes the test easier to read (at least in the case below).

Also the tests later on, are clearly testing different functions (like the shortenings). I think it would be rather confusing to have some test covering both functions and some covering only one.

Reply:
I see your point but I still prefer it to be split.
- As Davide said Code-duplication
- Easier to read and understand what the test are testing
- Less Unit Tests:
   - Imagine you have 3 different situations (empty-normal case-border-case). If you have N functions you need 3*N different Unit tests. I don't like it. I prefer to have at most N+3. 
- Easier to understand if they break. You simple read EmptyCallstack and understand that there is an issue there.

Reply:
As discussed:
Logically split the tests (by name) in separate tests for the two functions under test, moved some common constants above and shortened the test names automatically. See http://screenshot/7xh5YWJ2qKDqo8C